ui/src/lib/yjs/useYjsStore.ts

```diff
@@ -48,6 +48,16 @@ export default ({
     }
   }, [undoManager]);
 
+  const canUndo = useMemo(() => {
+    const stackLength = undoManager?.undoStack?.length ?? 0;
+    return stackLength > 0;
+  }, [undoManager?.undoStack?.length]);
+
+  const canRedo = useMemo(() => {
+    const stackLength = undoManager?.redoStack?.length ?? 0;
+    return stackLength > 0;
+  }, [undoManager?.redoStack?.length]);
```

Comment:
Please use these in handleWorkflowUndo/Redo in their respective conditional statements as well 🙏🏻

Reply:
You mean something like this? 

`const handleWorkflowUndo = useCallback(() => {
    const stackLength = undoManager?.undoStack?.length ?? 0;
    if (stackLength > 0) {
      undoManager?.undo();
    }
  }, [undoManager]);`